artiq/frontend/artiq_dashboard.py

```diff
@@ -108,6 +112,12 @@ def main():
         atexit.register(client.close_rpc)
         rpc_clients[target] = client
 
+    config = Client(args.server, args.port_control, "master_config")
+    atexit.register(config.close_rpc)
+    server_name = config.get_name()
+    if server_name is None:
+        server_name = args.server
```

Comment:
It's a bit tortuous with the (imprecise) test ``if server_name != args.server`` below. I would keep the initial value of this variable, and then do ``MainWindow(args.server if server_name is None else server_name)`` and test again ``server_name is None`` to generate the log message.

Reply:
Yep - that is much neater.